src/lambda_handler.py

```diff
@@ -34,54 +164,194 @@ def task_default(event):
+    # associated with an IP address
+    application_tag_name: str = event.get("application_tag", "Application")
+    # AWS resource tag name indicating whether an IP address should be published
+    publish_egress_tag_name: str = event.get("publish_egress_tag", "Publish Egress")
+    # Name of the IAM role to assume that can read the necessary EC2 data
+    # in each AWS account. Note that this role must exist in each account.
+    ec2_read_role_name: str = event.get("role_name", "EC2ReadOnly")
+
+    account_ids: List[str] = event.get("account_ids", None)
+    for account_id in account_ids:
+        # Verify AWS account ID is 12 digits
+        if not re.match(r"^\d{12}$", str(account_id)):
+            error_msg = 'Account ID "%s" is invalid - it must be 12 digits.'
+            logging.error(error_msg, account_id)
+            failed_task(result, error_msg % account_id)
+
+        logging.info("Examining account: %s" % account_id)
+
+        # Create an EC2 client with the assumed role
+        ec2: boto3.client = create_assumed_aws_client(
+            aws_service="ec2",
+            role_arn=f"arn:aws:iam::{account_id}:role/{ec2_read_role_name}",
+            session_name="publish-egress-ip-lambda",
+        )
+
+        # Get a list of all regions that match our filter
+        regions: List[str] = get_ec2_regions(ec2, region_filters)
+
+        logging.info("Gathering public IPs from %d regions" % len(regions))
+
+        # Loop through the region list and fetch the public EC2 IPs
+        for region in regions:
+            logging.info("Querying region: %s" % region)
+
+            # Create an EC2 resource with the assumed role in the specified
+            # region
+            ec2 = create_assumed_aws_resource(
+                aws_service="ec2",
+                region=region,
+                role_arn=f"arn:aws:iam::{account_id}:role/{ec2_read_role_name}",
+                session_name="publish-egress-ip-lambda",
+            )
+
+            # Get the public IPs of instances that are tagged to be published
+            for application_tag_value, public_ip in get_ec2_ips(
+                ec2, application_tag_name, publish_egress_tag_name
+            ):
+                # Loop through all regexes and add IP to set if matched
+                for config in file_configs:
+                    if config["app_regex"].match(application_tag_value):
+                        config["ip_set"].add(ip_network(public_ip))
+
+    # Use a single timestamp for all files
+    now = "{:%a %b %d %H:%M:%S UTC %Y}".format(datetime.utcnow())
+
+    # The bucket to publish the files to
+    bucket_name: str = event["bucket_name"]
+
+    # The domain to display in the header of each published file
+    domain: str = event.get("domain", "example.gov")
+
+    # Update each object (file) in the bucket
+    for config in file_configs:
+        # Initialize contents of object to be published
+        object_contents = "\n".join(file_header)
```

Comment:
With my proposed change we lost a `\n` at the end of the header.

```suggestion
        object_contents = "\n".join(file_header) + "\n"
```